pkg/build/builder/dockerutil.go

```diff
@@ -156,40 +157,43 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := client.RemoveContainer(docker.RemoveContainerOptions{ID: c.ID}); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
-	}()
-
-	// Start the container.
-	glog.V(4).Infof("Starting container %q ...", containerName)
-	if err := client.StartContainer(c.ID, nil); err != nil {
-		return fmt.Errorf("start container %q: %v", containerName, err)
 	}
 
-	// Stream container logs.
-	logsOpts.Container = c.ID
-	glog.V(4).Infof("Streaming logs of container %q with options %+v ...", containerName, logsOpts)
-	if err := client.Logs(logsOpts); err != nil {
-		return fmt.Errorf("streaming logs of %q: %v", containerName, err)
-	}
+	return interrupt.New(nil, removeContainer).Run(func() error {
```

Comment:
does not appear to:

```
I0331 23:00:14.457942       1 dockerutil.go:173] Starting container "openshift_s2i-build_ruby-sample-build-3_p1_post-commit" ...
I0331 23:00:14.720660       1 dockerutil.go:180] Streaming logs of container "openshift_s2i-build_ruby-sample-build-3_p1_post-commit" with options {Container:92fd0861d8531010703f53391ef5850ee1027e33147f44773a347f7d3d410019 OutputStream:0xc20802e008 ErrorStream:0xc20802e010 Follow:true Stdout:true Stderr:true Since:0 Timestamps:false Tail: RawTerminal:false} ...
/opt/rh/rh-ruby22/root/usr/bin/ruby -I"lib" -I"/opt/app-root/src/bundle/ruby/gems/rake-10.3.2/lib" "/opt/app-root/src/bundle/ruby/gems/rake-10.3.2/lib/rake/rake_test_loader.rb" "test/*_test.rb" 
Run options: --seed 28810

# Running:

.

Finished in 0.001928s, 518.5598 runs/s, 518.5598 assertions/s.

1 runs, 1 assertions, 0 failures, 0 errors, 0 skips
I0331 23:00:16.057152       1 dockerutil.go:186] Waiting for container "openshift_s2i-build_ruby-sample-build-3_p1_post-commit" to stop ...
I0331 23:00:16.623796       1 dockerutil.go:163] Removing container "openshift_s2i-build_ruby-sample-build-3_p1_post-commit" ...
```